File: ralff-be/src/main/java/org/example/utils/Logger.java
package org.example.utils;

public class Logger {
  private final boolean isEnabled;
  private final String name;


  public Logger(boolean isEnabled, String name) {
    this.isEnabled = isEnabled;
    this.name = name;
  }

  public Logger(boolean isEnabled) {
    this.isEnabled = isEnabled;
    this.name = null;
  }

  public void log(String... args) {
    System.out.println(args);
  }
}

File: ralff-fe/src/reverse-engineering/utils/logger.ts
export class Logger {
  private _enabled: boolean;
  private readonly name: string | undefined;

  constructor(enabled: boolean, name?: string) {
    this._enabled = enabled;
    this.name = name;
  }

  log(message?: any, ...optionalParams: any[]) {
    if (this._enabled) {
      const prefix = this.name ? `${this.name}:` : 'n/a';
      console.log(prefix, message, ...optionalParams);
    }
  }

  info(message?: any, ...optionalParams: any[]) {
    const prefix = this.name ? `${this.name}:` : 'n/a';
    console.log(prefix, message, ...optionalParams);
  }


  set enabled(value: boolean) {
    this._enabled = value;
  }
}